client/Containers/Login.jsx

```diff
@@ -8,47 +7,92 @@ class Login extends Component {
       email: '',
       password: '',
     };
+    this.onChange = this.onChange.bind(this);
+    this.onSubmit = this.onSubmit.bind(this);
+  }
+  onChange(ev) {
+    this.setState({ [ev.target.name]: ev.target.value });
+  }
+  onSubmit(ev) {
+    ev.preventDefault();
+    const { email, password } = this.state;
+    this.props.signIn({
+      email,
+      password,
+    });
   }
-
-  //componentDidUpdate
-  //perform axios call to authenticate user
-
-  handleSubmit = (e) => {
-    e.preventDefault();
-    console.log(this.state);
-    this.setState({ email: '', password: '' });
-  };
-
-  handleChange = ({ target: { name, value } }) => {
-    this.setState({ ...this.state, [name]: [value] });
-  };
```

Comment:
Arrow syntax inside of class components requires two babel plugins:
```
@babel/plugin-proposal-private-methods
@babel/plugin-proposal-class-properties
```
I'm personally a fan, because this binding inside of the constructor gets messy. These plugins also allow you to omit the constructor entirely and just declare state inside of the class block:
```javascript
class Login extends Component {
  state = {
    email: '',
    password: '',
  }
}
```

Reply:
good to know, i'll add these changes